React/CoreModules/RCTDevLoadingView.mm

```diff
@@ -266,16 +266,18 @@ - (void)showProgressMessage:(NSString *)message
 
 - (void)showOfflineMessage
 {
-  RCTUIColor *color = [RCTUIColor whiteColor]; // TODO(macOS GH#774)
-  RCTUIColor *backgroundColor = [RCTUIColor blackColor]; // TODO(macOS GH#774)
+  dispatch_async(dispatch_get_main_queue(), ^{
+    RCTUIColor *color = [RCTUIColor whiteColor]; // TODO(macOS GH#774)
+    RCTUIColor *backgroundColor = [RCTUIColor blackColor]; // TODO(macOS GH#774)
 
-  if ([self isDarkModeEnabled]) {
-    color = [RCTUIColor blackColor]; // TODO(macOS GH#774)
-    backgroundColor = [RCTUIColor whiteColor]; // TODO(macOS GH#774)
-  }
+    if ([self isDarkModeEnabled]) {
```

Comment:
For my own sake, this is the line that eventually calls `RCTColorSchemePreference(nil)` (Line 289 below), which is checking NSAppearance on a background thread?

Reply:
If so, you could maybe also override `requiresMainQueueSetup` to return YES on line 70 ?

Reply:
what caused that change? i would assume checking isdarkmode on ios in the background thread is also wrong. why is it only macOS issue?

Reply:
I'd have to defer to @tido64 on this one.